modules/unsupported/process-feature/src/main/java/org/geotools/process/vector/ClipProcess.java

```diff
@@ -292,6 +281,19 @@ private Object clipGeometry(Geometry geom, Class target, CoordinateReferenceSyst
 
             // manage Z preservation
             if (preserveZ && !geom.equalsExact(clipped)) {
+                // Check if the clipped geometry has 3 ordinates across entire area.
```

Comment:
Is there a unit test covering this?

Reply:
Yes, it was failing hence the reason for the fix.

We actually found this fix when upgrading to 1.16 and introduced some code in CoordinateArraySequence to "fix" bad arrays. However that slowed down things *everywhere* rather than this one spot which is the only known example where we end up mixing coordinates.